subworkflows/local/prepare_genome.nf

```diff
@@ -27,110 +27,73 @@ workflow PREPARE_GENOME {
         germline_resource // channel: [optional]  germline_resource
         known_indels      // channel: [optional]  known_indels
         pon               // channel: [optional]  pon
-        tools             // value:   [mandatory] tools
-        step              // value:   [mandatory] step
 
     main:
 
-    if(!tools) tools = ""
-
     ch_versions = Channel.empty()
 
-    ch_bwa = Channel.empty()
-    if (!(params.bwa) && 'mapping' in step) {
-        if (params.aligner == "bwa-mem") {
-            BWAMEM1_INDEX(fasta)
-            ch_bwa = BWAMEM1_INDEX.out.index
-            ch_versions = ch_versions.mix(BWAMEM1_INDEX.out.versions)
-        } else if (params.aligner == "bwa-mem2") {
-            BWAMEM2_INDEX(fasta)
-            ch_bwa = BWAMEM2_INDEX.out.index
-            ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)
-        }
-    }
+    (ch_bwa1, ch_bwa_version) = BWAMEM1_INDEX(fasta)
+    (ch_bwa2, ch_bwa_version) = BWAMEM2_INDEX(fasta)
```

Comment:
won't `ch_bwa_version` be overwritten here?

Reply:
I mean only one of them will be run, if the logic is well in modules.config, but just wondering for weird effects...

Reply:
Yes, you're correct.
`ch_bwa_version` will only ever be the output of `BWAMEM2_INDEX(fasta)` from these lines.

Reply:
meant to fix it as I did with `ch_bwa1` and `ch_bwa2`, but forgot...
I'm on it, thanks

Reply:
I don't like using `PROCESS.out` much, but if you prefer it this way, why not

Reply:
It depends what's more maintainable for you. 

I would recommend `PROCESS.out` in general as it's easier to see (at least that is my belief), and doesn't create extra channels that are easy to make mistakes with (e.g. doing the above).

But if it's more maintainable to reassign channels, then keep doing it.